xds/internal/xdsclient/xdsresource/unmarshal_eds_test.go

```diff
@@ -333,6 +335,120 @@ func (s) TestEDSParseRespProtoAdditionalAddrs(t *testing.T) {
 	}
 }
 
+func (s) TestUnmarshalEndpointHashKey(t *testing.T) {
```

Comment:
Here as well. Let's have a test case where the env var is set to `true` and ensure that we don't populate the `HashKey` field in the parsed resource even if it is present in the EDS resource.

Reply:
Nit: could we please set a different value for the `name` field in the test table though. Thanks.